### PR TITLE
stupidlogger

### PR DESCRIPTION
**Pull request by k323r**
This logging service simply logs all incoming data in json format.

### CODE REVIEW

**Comment by k323r** [comment]
Still missing: 
- raw unpacker that just return json
- stupid logger service that subscribes to zmq with unpacker = raw

File: msb/network/pubsub/factories.py [hunk 1]
```diff
@@ -55,6 +55,7 @@ def register_subscriber(name, subscriber_class: Subscriber, config):
 
 register_subscriber("zmq", ZMQ_Subscriber, ZMQConf)
 register_subscriber("mqtt", MQTT_Subscriber, MQTTConf)
+register_subscriber("zmq_raw", ZMQ_Subscriber, ZMQConf)
```
Comment (kawindme) [inline msb/network/pubsub/factories.py:58]: Why register a new subscriber under a new name but with the same parameters?

File: msb/network/zmq/publisher.py [hunk 2]
```diff
@@ -14,7 +15,7 @@ def __init__(self, config: ZMQConf):
         self.context = zmq.Context.instance()
         self.socket = self.context.socket(zmq.PUB)
 
-        self.packer = json
+        self.packer = get_packer(config.packstyle)
```
Comment (kawindme) [inline msb/network/zmq/publisher.py:18]: I think this is named `packer` now in the config. I don't have a clear preference whether we should use packer or packstyle.

File: msb/network/zmq/publisher.py [hunk 3]
```diff
@@ -25,7 +26,7 @@ def connect(self):
             sys.exit(-1)
 
     def send(self, topic: bytes, data: dict):
-        data = self.packer.dumps(data)
+        data = self.packer(data)
```
Comment (kawindme) [inline msb/network/zmq/publisher.py:29]: If `packer` is a function like `json.dumps` the name `packer` is misleading.

File: msb/network/zmq/subscriber.py [hunk 4]
```diff
@@ -17,9 +17,7 @@ def __init__(self, topic: bytes, config: ZMQConf):
         self.connect()
         self.subscribe(topic)
 
-        # Possible to add a switch via config here
-        # unpacker = pickle
-        self.unpacker = json
+        self.unpacker = get_unpacker(config.packstyle)
```
Comment (kawindme) [inline msb/network/zmq/subscriber.py:20]: Same as for the Publisher, `packstyle` is not defined in config.

File: msb/network/zmq/subscriber.py [hunk 5]
```diff
@@ -50,7 +48,7 @@ def receive(self) -> tuple[bytes, dict]:
             tuple(topic: bytes, message: dict): the message received
         """
         (topic, message) = self.socket.recv_multipart()
-        message = self.unpacker.loads(message.decode())
+        message = self.unpacker(message.decode())
```
Comment (kawindme) [inline msb/network/zmq/subscriber.py:51]: Same as for publisher, `unpacker` is a misleading name for a function.

File: msb/stupidlog/msb_stupidlog.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+from msb.network import get_subscriber
+from msb.stupidlog.config import StupidlogConf
+
+def signal_handler(sig, frame):
+    print("msb_fusionlog.py exit")
+
+    sys.exit(0)
+
+
+class FusionlogService:
```
Comment (kawindme) [inline msb/stupidlog/msb_stupidlog.py:14]: Name

File: msb/stupidlog/msb_stupidlog.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        for topic, data in self.get_data():
+            if self.config.print_stdout:
+                print(f"{topic} : {data}")
+            # log here
+
+
+def main():
+    signal.signal(signal.SIGINT, signal_handler)
+    config = load_config(StupidlogConf(), "stupidlog")
+    subscriber = get_subscriber('zmq', topic=config.topics)
```
Comment (kawindme) [inline msb/stupidlog/msb_stupidlog.py:39]: The newly defined `zmq_raw` subscriber is not even used.

File: msb/stupidlog/msb_stupidlog.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+class FusionlogService:
+    def __init__(self, config: FusionlogConf, subscriber: Subscriber):
+        self.config = config
+        self.subscriber = subscriber
+
+    def get_data(self):
+        while True:
+            try:
+                (topic, data) = self.subscriber.receive()
+            except Exception as e:
```
Comment (kawindme) [inline msb/stupidlog/msb_stupidlog.py:23]: Specify exception subclass.

File: msb/stupidlog/msb_stupidlog.py (hunk 7 above)
Comment (kawindme) [reply msb/stupidlog/msb_stupidlog.py:39]: The unpackstyle `"raw"` is not used either.

File: requirements.txt [hunk 9]
```diff
@@ -12,3 +12,5 @@ quaternion==3.5.2.post4
 # scipy==1.10.1
 smbus==1.1.post2
 uptime==3.0.1
+adafruit-circuitpython-lps2x
```
Comment (kawindme) [inline requirements.txt:15]: These requirements were added for the dew service. They should not be added here.

File: msb/stupidlog/msb_stupidlog.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+                print(f"failed to receive message: {e}")
+                continue
+            topic = topic.decode("utf-8")
+            yield topic, data
+
+    def run(self):
+        for topic, data in self.get_data():
+            if self.config.print_stdout:
+                print(f"{topic} : {data}")
+            # log here
```
Comment (kawindme) [inline msb/stupidlog/msb_stupidlog.py:33]: Logging not yet implemented (?)

**Review by kawindme — CHANGES_REQUESTED** [review]
The logger does not seem to be implemented yet. 
The name `Stupidlogger` is rather, well – stupid. 
If the purpose is to log raw JSON data I would prefer one of the following names:
- `RawLogger`
- `JSONLogger`
- `SimpleLogger`
- ...

How do you plan to save the raw JSON's? One file per time step?
I would assume that would mean too many writes for the disk / SD-card to handle.

File: msb/network/pubsub/factories.py (hunk 1 above)
Comment (k323r) [reply msb/network/pubsub/factories.py]: this was the initial idea to solve the packstyle-problem - removed it

File: msb/network/zmq/publisher.py (hunk 2 above)
Comment (k323r) [reply msb/network/zmq/publisher.py:18]: i think this should be named packstyle and unpackstyle respectivel

File: msb/network/zmq/publisher.py (hunk 3 above)
Comment (k323r) [reply msb/network/zmq/publisher.py:29]: good point, I'd prefer `pack` and `unpack` as names

File: msb/stupidlog/msb_stupidlog.py (hunk 7 above)
Comment (k323r) [reply msb/stupidlog/msb_stupidlog.py]: removed it, was just an idea. solved in msb_simplelog.py by overriding the packer

**Pull request closed without merging**